openfecwebapp/data_mappings.py

```diff
@@ -0,0 +1,195 @@
+import re
+import locale
+
+from flask import url_for
+
+locale.setlocale(locale.LC_ALL, '')
+
+# template vars for pagination results counts and
+# next/prev links on tabular views
+def generate_pagination_values(c, params, url, data_type):
+    pagination = {}
+    per_page = int(c['pagination']['per_page'])
+    page = int(c['pagination']['page'])
+    current_results_start = per_page * (page - 1) + 1 
+    current_results_end = per_page * page
+    total_pages = int(c['pagination']['pages'])
+
+    pagination['results_count'] = c['pagination']['count']
+    pagination['page'] = page
+    pagination['per_page'] = per_page
+    pagination['current_results_start'] = current_results_start
+    pagination['current_results_end'] = current_results_end
+
+    if current_results_start or current_results_end:
+        pagination['results_range'] = True
+
+    # next url
+    if page < total_pages:
+        next_page_num = str(page + 1)
+        params['page'] = next_page_num
+        pagination['next_url'] = url_for(data_type, **params) 
+        pagination['pagination_links'] = True
+
+    # prev url
+    if page - 1 > 0:
+        prev_page_num = str(page - 1)
+        params['page'] = prev_page_num
+        pagination['prev_url'] = url_for(data_type, **params) 
+        pagination['pagination_links'] = True
+
+    return pagination
+
+# maps basic values for candidates
+def map_candidate_table_values(c):
+    candidate = {
+        'name': c['name']['full_name'],
+        'office': c['elections'][0]['office_sought_full'],
+        'election': int(c['elections'][0]['election_year']),
+        'party': c['elections'][0]['party_affiliation'],
+        'state': c['elections'][0]['state'],
+        'district': int(c['elections'][0]['district']) 
+            if c['elections'][0]['district'] else '',
+        'nameURL': '/candidates/' + c['candidate_id'],
+        'id': c['candidate_id']
+    }
+
+    return candidate
+
+# maps basic values for committees
+def map_committee_table_values(c):
+    committee = {}
+
+    if c.get('description'):
+        committee['name'] = c['description'].get('name', '')
+        committee['organization'] = c['description'].get(
+            'organization_type_full', '')
+
+    if c.get('treasurer'):
+        committee['treasurer'] = c['treasurer'].get('name_full', '')
+
+    if c.get('address'):
+        committee['state'] = c['address'].get('state')
+
+    if c.get('status'):
+        committee['type'] = c['status'].get('type_full', '')
+        committee['designation'] = c['status'].get(
+            'designation_full', '')
+
+    committee['nameURL'] ='/committees/' + c.get('committee_id', '')
```

Comment:
Good call. AFAIK, the *only* thing we can be sure will exist is the ID, which is why I was lax. You're probably right and I should tighten this up.